includes/validation/class-amp-validation-error-taxonomy.php

```diff
@@ -2872,12 +2857,14 @@ private static function is_validation_error_for_js_script_element( $validation_e
 	/**
 	 * Get Error Title from Code
 	 *
+	 * @todo The message here should be constructed in the sanitizer that emitted the validation error in the first place.
+	 *
 	 * @param array $validation_error Validation error.
 	 * @return string Title with some formatting markup.
 	 */
 	public static function get_error_title_from_code( $validation_error ) {
 		switch ( $validation_error['code'] ) {
-			case self::INVALID_ELEMENT_CODE:
+			case AMP_Tag_And_Attribute_Sanitizer::DISALLOWED_TAG:
```

Comment:
Yes, I like this a lot. I think something like that should be the end state. I think it is bound up with revisiting the spec generation into PHP objects (#3730). We need a way to easily connect validation errors with their related specs (#3817). This will also be key for making a reusable PHP package for other platforms to use (#2315).

So I think keeping the current approach of passing around is a good interim step in the plugin's evolution, but we should work toward what you propose here.

Reply:
I created a new issue for the above: https://github.com/ampproject/amp-wp/issues/3875